src/jit/lowerxarch.cpp

```diff
@@ -2296,6 +2296,47 @@ void Lowering::ContainCheckSIMD(GenTreeSIMD* simdNode)
 #endif // FEATURE_SIMD
 
 #ifdef FEATURE_HW_INTRINSICS
+//----------------------------------------------------------------------------------------------
+// IsContainableHWIntrinsicOp: Return true if 'node' is a containable HWIntrinsic op.
+//
+//  Arguments:
+//     categoryOfContainingNode - The category of the node for which 'node' is an op.
+//     node - The hardware intrinsic node.
+//
+bool Lowering::IsContainableHWIntrinsicOp(HWIntrinsicCategory categoryOfContainingNode, GenTree* node)
+{
+    if (!node->OperIsHWIntrinsic())
+    {
+        return false;
+    }
+
+    // TODO-XArch: Update this to be table driven, if possible.
+
+    NamedIntrinsic intrinsicID = node->AsHWIntrinsic()->gtHWIntrinsicId;
+
+    switch (intrinsicID)
+    {
+        // Non-VEX encoded instructions require aligned memory ops, so we can fold them.
+        // However, we cannot do the same for the VEX-encoding as it changes an observable
+        // side-effect and may mask an Access Violation that would otherwise occur.
+        case NI_SSE_LoadAlignedVector128:
+            return (category == HW_Category_SimpleSIMD) && !comp->canUseVexEncoding();
+
+        // Only fold a scalar load into a SIMD scalar intrinsic to ensure the number of bits
+        // read remains the same. Likewise, we can't fold a larger load into a SIMD scalar
+        // intrinsic as that would read fewer bits that requested.
+        case NI_SSE_LoadScalar:
```

Comment:
Probably want to assert here that the baseType of the containing node is the same as the baseType of the load.

It should already line up due to the method signatures, but we don't want to end up in a scenario where the `LoadScalar` is for m32, but the containing node would do a m64.

Reply:
Fixed.